pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -462,10 +462,19 @@ func (m *rangefeedMuxer) restartActiveRangeFeed(
 		return err
 	}
 
-	if errInfo.resolveSpan {
-		return divideSpanOnRangeBoundaries(ctx, m.ds, rs, active.startAfter, m.startSingleRangeFeed)
-	}
-	return m.startSingleRangeFeed(ctx, rs, active.startAfter, active.token)
+	// Restart rangefeed on another goroutine.  There are two reasons for doing this
+	// under muxer context group.
+	//   1. Restart might be a bit expensive, particularly if we have to resolve span.
+	//      We do not want to block receiveEventsFromNode for too long.
+	//   2. Most importantly: the context passed to this function may be canceled if
+	//      the restart is happening because node mux is exiting.
+	m.g.GoCtx(func(ctx context.Context) error {
```

Comment:
Drive-by: isn't there a risk here that we'll spawn a bunch of goroutines whenever a rangefeed fails (see e.g. the `closeWithRestart` call), and isn't that kind of goroutine burst exactly what we're trying to avoid with the mux rangefeeds?

Reply:
sigh... yeah... And.. I don't even know if this is a problem... like, sure, ctx will get canceled
shortly.  But each of those streams must be restarted against another mux node... Or new connect (with new context) must be established.  I really need to recreate this deadlock, but no such luck so far.